sickbeard/providers/btdigg.py

```diff
@@ -0,0 +1,181 @@
+from sickbeard.exceptions import ex
+
+class BTDIGGProvider(generic.TorrentProvider):
+    def __init__(self):
+		generic.TorrentProvider.__init__(self, "BTDigg")
+		
+		self.supportsBacklog = True
+		self.session = requests.Session()
+		self.url = 'https://api.btdigg.org/'
+		
+    def isEnabled(self):
+        return self.enabled
+    
+    
+    def imageName(self):
+        return 'btdigg.png'
+    
+
+    # def getQuality(self, item):        
+
+        # quality = Quality.sceneQuality(item[0], anime)
+        # return quality
+
+
+    def _get_airbydate_season_range(self, season):        
+        if season == None:
+            return ()        
+        year, month = map(int, season.split('-'))
+        min_date = datetime.date(year, month, 1)
+        if month == 12:
+            max_date = datetime.date(year, month, 31)
+        else:    
+            max_date = datetime.date(year, month+1, 1) -  datetime.timedelta(days=1)
+        return (min_date, max_date)    
+
+
+    def _get_season_search_strings(self, show, season=None):
+        search_string = []
+    
+        if not show:
+            return []
+      
+        myDB = db.DBConnection()
+        
+        if show.air_by_date:
+            (min_date, max_date) = self._get_airbydate_season_range(season)
+            sqlResults = myDB.select("SELECT * FROM tv_episodes WHERE showid = ? AND airdate >= ? AND airdate <= ?", [show.tvdbid,  min_date.toordinal(), max_date.toordinal()])
+        else:
+            sqlResults = myDB.select("SELECT * FROM tv_episodes WHERE showid = ? AND season = ?", [show.tvdbid, season])
+            
+        for sqlEp in sqlResults:
+            if show.getOverview(int(sqlEp["status"])) in (Overview.WANTED, Overview.QUAL):
+                if show.air_by_date:
+                    for show_name in set(show_name_helpers.allPossibleShowNames(show)):
+                        ep_string = show_name_helpers.sanitizeSceneName(show_name) +' '+ str(datetime.date.fromordinal(sqlEp["airdate"])).replace('-', '.')
+                        search_string.append(ep_string)
+                else:
+                    for show_name in set(show_name_helpers.allPossibleShowNames(show)):
+                        ep_string = show_name_helpers.sanitizeSceneName(show_name) +' '+ sickbeard.config.naming_ep_type[2] % {'seasonnumber': season, 'episodenumber': int(sqlEp["episode"])}
+                        search_string.append(ep_string)                       
+        return search_string
+
+
+    def _get_episode_search_strings(self, ep_obj, add_string=''):
+        
+        search_string = []
+       
+        if not ep_obj:
+            return []
+        if ep_obj.show.air_by_date:
+            for show_name in set(show_name_helpers.allPossibleShowNames(ep_obj.show)):
+                ep_string = show_name_helpers.sanitizeSceneName(show_name) +' '+ str(ep_obj.airdate).replace('-', '.')
+                search_string.append(ep_string)
+		if len(add_string):
```

Comment:
bad indent, also this if and the += line need to be just above `search_string.append(ep_string)` not under it.